feat: create export actions with csv format (#1022)
graasp/graasp: src/services/item/plugins/action/specifications.md

# Item actions

> An item has a related actions grassped from different views (Builder, Player, ...etc)

Reviewer: ```suggestion
An item is linked to actions captured across different views (Builder, Player, ...etc).
```

> The action basically has the following proporites

Reviewer: ```suggestion
The action basically has the following properties
```

- `id`

> - `view`: The context in which the action is viewed.

Reviewer: ```suggestion
- `view`: The service (builder, player, library) in which the action is captured.
```

> - `type`: types listed within sdk `ActionTriggers`.

Reviewer: Give a few examples, and add a link to the code in sdk.

- `extra`: this field stores additional data relevant to the specific action.
- `item`: the related item.

## Endpoints

- `GET /items/:id/actions`: Retrieves a list of actions associated with a specific item.
- `GET /items/:id/actions/aggregation`: Fetches aggregated data for actions corresponding to the specified item id.
- `POST /items/:id/actions`: save item actions.

> - `POST /items/:id/actions/export`: Exports actions tied to a particular item. The system sends the generated files to the user via email. Users are allowed to receive an archive in two formats once a week:

Reviewer: The frequency is once every 24 hours if I remember correctly. Was this changed ?

>   - `json` ( default format)

Reviewer: ```suggestion
  - `json` (default format)
```

>   - `csv` we use `papaParse` For CSV files, we use papaParse to convert JSON into CSV format. To address the challenge of nested objects, we flatten them as shown below:

Reviewer: I am not sure we need to specify the library used to write the csv file.

  ```js
  const item = {
    id: 'id',
    namd: 'name',
    creator: {
      id: 'member-id',
      name: 'member-name',
    },
  };

  const flattenItem = {
    id: 'id',
    namd: 'name',
    'creator.id': 'member-id',
    'creator.name': 'member-name',
  };

>   ```

Reviewer: These examples are nice, maybe you can show the first one in JSON format and the second one in csv format ?
